.github/workflows/githubci.yml

```diff
@@ -0,0 +1,32 @@
+name: Arduino Library CI
+
+on: [pull_request, push, repository_dispatch]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - uses: actions/checkout@v2
+    - uses: actions/checkout@v2
+      with:
+         repository: adafruit/ci-arduino
+         path: ci
+
+    - name: pre-install
+      run: bash ci/actions_install.sh
+
+    - name: test platforms
+      run: python3 ci/build_platform.py main_platforms
+
+    - name: clang
+      run: python3 ci/run-clang-format.py -e "ci/*" -e "bin/*" -r . 
+
+    - name: doxygen
+      env:
+        GH_REPO_TOKEN: ${{ secrets.GH_REPO_TOKEN }}
+        PRETTYNAME : "RTClib"
+      run: bash ci/doxy_gen_and_deploy.sh
```

Comment:
This script expects to be run in a Travis build environment, and makes use of some Travis-specific environment variables. See for example commit fae1b6904b198a92e5133d159885cd801ee2ccaf: the author is incorrectly set to “travis@travis-ci.org”, and the commit message is incomplete. The script also fails to check whether the commit lands on master, thus if someone works on a feature branch without using a fork, the Doxygen docs for the feature branch will be published and replace the master docs.